File: recursion/go/Recursion5.go
package main

import "fmt"

// first n natural number
func printFirstNOddNaturalNumber(number int) {
	// negative number
	if number < 1 {
		return
	}

	// base case
	if number == 1 {
		fmt.Print(number, ", ")
		return
	}

	// start with Odd 
if(number % 2 == 0){
	 number = number -1
}
	// recursive call
	printFirstNOddNaturalNumber(number - 2)
	fmt.Print(number, ", ")

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNOddNaturalNumber(number)
}

File: recursion/java/Recursion5.java
import java.util.Scanner;

public class Recursion5 {

    public static void printFirstNOddNaturalNumber(int n) {
        // negative number
        if (n < 1) {
            return;
        }

        if (n == 1) {
            System.out.print(n + ", ");
            return;
        }
        // make Odd
        if (n % 2 == 0) {
            n = n - 1;
        }

        printFirstNOddNaturalNumber(n - 2);
        System.out.print(n + ", ");

    }

    public static Scanner scan = new Scanner(System.in);

    public static void main(String... args) {
        int n = scan.nextInt();
        printFirstNOddNaturalNumber(n);
    }
}